### PR TITLE
[ARROW] LocalTableScanExec should not trigger job

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

Before this PR:

截屏2023-04-14 下午5 19 52

After this PR:

截屏2023-04-14 下午5 18 16

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala [hunk 1]
```diff
@@ -51,6 +51,8 @@ object SparkDatasetHelper extends Logging {
       doCollectLimit(collectLimit)
     case collectLimit: CollectLimitExec if collectLimit.limit < 0 =>
       executeArrowBatchCollect(collectLimit.child)
+    case localTableScan: LocalTableScanExec =>
+      doLocalTableScan(localTableScan)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala:55]: it seems we need to post `SparkListenerDriverAccumUpdates` to update metrics

**Review by ulysses-you — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala (hunk 1 above)
Comment (cfmcgrady) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala:55]: Spark also doesn't post the driver-side SQLMetrics, is that a Spark bug?

the SQL UI will be changed once we post the metrics.

Before:

截屏2023-04-18 下午2 37 50


After:

截屏2023-04-18 下午2 37 04

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala (hunk 1 above)
Comment (cfmcgrady) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala:55]: filed a PR on https://github.com/apache/spark/pull/40875, and will send a follow-up once Spark has addressed this issue.

**Pull request closed without merging**